### PR TITLE
Icons: Move unicons icons to the repository and generate the iconsBundle.js with nodejs

### PR DESCRIPTION
**Pull request by academo**
**😮 Note to PR reviewers:** This PR doesn't change ~1400 files. This big number is due to the more than 1000 icons that are moved from https://github.com/grafana/icons/ to here.

> Tip: disable SVG files in github file filter: 
image

* Moves the unicons currently stored in a [separate repo](https://github.com/grafana/icons/) to the grafana main repository.
* Modifies the licensing files to keep these icons in the Apache2 License.
* Introduces a JS version of the [go script](https://github.com/grafana/icons/blob/main/cmd/genbundle/main.go) previously used to generate the `iconsBundle.ts` file
* Removes the `copy-icons` script and post install script since they are now located in the correct folder.
* Uses a custom script to generate the `iconsBundle.ts`. This can be run from `@grafana/ui` as `yarn generate-icons-bundle-cache-file` the list of cached files can be found at `@grafana/ui/src/components/Icon/cached.json`
* Removes `iconscout-unicons-tarball` from package.json as is not longer necessary.
* Monochromes icons are not moved since they are not used anywhere.

NOTE: These changes don't change bundle sizes in anyway, the amount of cached icons and the way they are cached remains the same. The icons were already present in `public/img/icons` on build time but brought in using a different method. This PR moves them there directly as part of the repo to remove that process.

### CODE REVIEW

File: packages/grafana-ui/package.json [hunk 1]
```diff
@@ -36,7 +36,8 @@
     "docsExtract": "mkdir -p ../../reports/docs && api-extractor run 2>&1 | tee ../../reports/docs/$(basename $(pwd)).log",
     "storybook": "start-storybook -p 9001 -c .storybook",
     "storybook:build": "build-storybook -o ./dist/storybook -c .storybook",
-    "typecheck": "tsc --emitDeclarationOnly false --noEmit"
+    "typecheck": "tsc --emitDeclarationOnly false --noEmit",
+    "generate-icons-bundle": "node ./scripts/generate-icon-bundle.js"
```
Comment (academo) [inline packages/grafana-ui/package.json:40]: new script to generate the iconsBundle.ts file.

Comment (academo) [inline packages/grafana-ui/package.json:40]: New script to generate the iconBundle.ts file

File: packages/grafana-ui/scripts/generate-icon-bundle.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+const fs = require('fs');
```
Comment (academo) [inline packages/grafana-ui/scripts/generate-icon-bundle.js:1]: Aims to replace https://github.com/grafana/icons/blob/main/cmd/genbundle/main.go

File: packages/grafana-ui/src/icons/cached.txt [hunk 3]
```diff
@@ -0,0 +1,155 @@
+unicons/angle-double-down
```
Comment (academo) [inline packages/grafana-ui/src/icons/cached.txt:1]: list taken from https://github.com/grafana/icons/blob/main/cmd/genbundle/bundle.txt

**Comment by ryantxu** [comment]
Before... we download a 300kb zip that is extracted into:
image

This now has these icons directly in the git tree:
image

Looks like we can remove "monochrome"...

File: packages/grafana-ui/src/components/Icon/cached.json [hunk 4]
```diff
@@ -0,0 +1,157 @@
+[
```
Comment (academo) [inline packages/grafana-ui/src/components/Icon/cached.json:1]: list taken from [https://github.com/grafana/icons/blob/main/cmd/genbundle/bundle.txt](https://github.com/grafana/icons/blob/main/cmd/genbundle/bundle.txt?rgh-link-date=2022-08-16T11%3A13%3A55Z)

**Comment by academo** [comment]
> I'm not sure the best solution here -- IIUC, this will add 6mb of icons to the grafana-ui bundle... is that what we want? Should we just put them in `/public/static/...`?
> 
> Given that it is loaded as a runtime dependency in plugins that probably does not matter, but something we should be careful of.

@ryantxu  These changes themselves won't add anything to the bundle, they only change the way icons are pulled to the build.

The icons caching and list remains the same, therefore no change in how many or how plugins are cached.

Putting the icons in public doesn't affect the build or bundle as they were already there just not directly in the repo but brought from a third party dependency and a script that puts them there.

**Comment by academo** [comment]
> Before... we download a 300kb zip that is extracted into: image
> 
> This now has these icons directly in the git tree: image
> 
> Looks like we can remove "monochrome"...

I agree. I don't see we are using monochrome anywhere. I'll remove them.

**Review by ryantxu — APPROVED** [review]
I think this also needs to remove the .gitignore file under:

image

**Comment by ryantxu** [comment]
After this change, we should update the README in the icons repo and likely archive the repo.  It is still needed for the 8x branch and older builds, but does not need to be updated

**Comment by academo** [comment]
> I think this also needs to remove the .gitignore file under:
> 
> image

Strange. I deleted it and commit the changes before you wrote this comment. [see here](https://github.com/grafana/grafana/pull/53766/files?file-filters%5B%5D=dotfile&show-deleted-files=true&show-viewed-files=true) 

image

**Review by jackw — APPROVED** [review]
LGTM! 🚀

**Pull request merged**